phoenix-core-client/src/main/java/org/apache/phoenix/jdbc/HighAvailabilityGroup.java

```diff
@@ -195,36 +198,104 @@ private HighAvailabilityGroup(HAGroupInfo info, Properties properties) {
         this.state = state;
     }
 
-    public static HAGroupInfo getHAGroupInfo(String url, Properties properties)
-            throws SQLException {
-        if (url.startsWith(PhoenixRuntime.JDBC_PROTOCOL)) {
-            url = url.substring(PhoenixRuntime.JDBC_PROTOCOL.length() + 1);
-        }
-        if (!(url.contains("[") && url.contains("|") && url.contains("]"))) {
-            throw new SQLExceptionInfo.Builder(SQLExceptionCode.MALFORMED_CONNECTION_URL)
-                    .setMessage(String.format("URL %s is not a valid HA connection string", url))
-                    .build()
-                    .buildException();
-        }
+    /**
+     * Get an instance of HAURLInfo given the HA connecting URL (with "|") and client properties.
+     * Here we do parsing of url and try to extract principal and other additional params
+     * @throws SQLException
+     */
+    public static HAURLInfo getUrlInfo(String url, Properties properties) throws SQLException {
```

Comment:
Are there unit tests for various Url patterns and use cases?

Reply:
I have added tests with different principles, I think can add more with without principle as well